pkg/adapter/awssnssource/adapter.go

```diff
@@ -95,31 +94,16 @@ const (
 
 // Start implements adapter.Adapter.
 func (a *adapter) Start(ctx context.Context) error {
-	// ctx gets canceled to stop goroutines
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	// handle stop signals
-	go func() {
-		<-ctx.Done()
-		a.logger.Info("Shutdown signal received. Terminating")
-		cancel()
-	}()
-
 	http.HandleFunc("/", a.handleNotification)
 	http.HandleFunc("/health", healthCheckHandler)
 
 	server := &http.Server{Addr: ":" + serverPort}
 	serverErrCh := make(chan error)
 	defer close(serverErrCh)
 
-	var wg sync.WaitGroup
-
-	wg.Add(1)
```

Comment:
No need for a WaitGroup when there is a single goroutine to wait for and that goroutine already sends its result to a channel (`serverErrCh` in this case)